### DISCUSSION TITLE
How to overlay one component on top of another

### DISCUSSION THREAD

**Discussion by shalldie** [charmbracelet/lipgloss#248] [Q&A] [answered]
I want to create a `Dialog` component, which overlay whole app.

After check the example code in `layout.go`, I found the methods `lipgloss.Place` + `lipgloss.WithWhitespaceChars` may create a new block. Is there any way to overlay a block instead of create it ?

Like this:

<img width="682" alt="image" src="https://github.com/charmbracelet/lipgloss/assets/9987486/3401fdad-0d4a-4635-a8f0-99432e1ced2a">

**Comment by BigJk** [comment, marked as answer]
I also wondered this a while ago. Because of one of the screenshots on the charm page I thought that it might be implemented somewhere already, which isn't the case. You can check the code of this PR for a solution: https://github.com/charmbracelet/lipgloss/pull/102

**Reply by shalldie** [reply]
Thanks, helps a lot! I try to edit the code to make some hacky and it works.
